doc/tutorials/acrn-secure-boot-with-grub.rst

```diff
@@ -0,0 +1,256 @@
+    sleep 10
+    reboot
+
+Replace the ESP_UUID with the UUID of your EFI system partition (found
+by running the :command:`lsblk -f`.  In the example output below,
+the UUID is ``24FC-BE7A``:
+
+.. code-block:: console
+    :emphasize-lines: 2
+
+    sda
+    ├─sda1 vfat   ESP    24FC-BE7A                            /boot/efi
+    ├─sda2 vfat   OS     7015-557F
+    ├─sda3 ext4   UBUNTU e8640994-b2a3-45ad-9b72-e68960fb22f0 /
+    └─sda4 swap          262d1113-64be-4910-a700-670b9d2277cc [SWAP]
+
+
+Enable Authentication in GRUB
+=============================
+
+With authentication enabled, a user/password is required to restrict
+access to the GRUB shell, where arbitrary commands could be run.
+A typical GRUB configuration fragment (added to ``grub.init.cfg``) might
+look like this::
+
+    set superusers="root"
+    export superusers
+    password_pbkdf2 root GRUB_PASSWORD_HASH
+
+Replace the ``GRUB_PASSWORD_HASH`` with the result of the :command:`grub-mkpasswd-pbkdf2`
+with your custom passphrase.
+
+Use this command to sign the :file:`grub.init.cfg` file with your private
+GPG key and create the :file:`grub.init.cfg.sig`::
+
+    gpg --homedir keys --detach-sign grub.init.cfg
+
+
+Create a single GRUB EFI Binary
+===============================
+
+Use the ``grub-mkstandalone`` tool to create a single GRUB EFI binary
+file with the buit-in modules and the signed ``grub.init.cfg`` file.
+The ``--pubkey`` option adds a GPG public key that will be used for
+verification. The public key ``boot.key`` is no longer required.
+
+.. note::
+    You should make a backup copy of your current GRUB image
+    (:file:`grubx64.efi`) before replacing it with the new signed GRUB image.
+    This would allow you to restore GRUB in case of errors updating it.
+
+Here's an example script to do this build::
+
+    #!/bin/bash
+    #
+
+    TARGET_EFI='/boot/efi/EFI/ubuntu/grubx64.efi'
+
+    # GRUB doesn't allow loading new modules from disk when secure boot is in
+    # effect, therefore pre-load the required modules.
+
+    MODULES="all_video archelp boot bufio configfile crypto echo efi_gop efi_uga ext2 extcmd  \
+     fat font fshelp gcry_dsa gcry_rsa gcry_sha1 gcry_sha512 gettext gfxterm linux linuxefi ls \
+     memdisk minicmd mmap mpi normal part_gpt part_msdos password_pbkdf2 pbkdf2 reboot relocator \
+     search search_fs_file search_fs_uuid search_label sleep tar terminal verifiers video_fb"
+
+    grub-mkstandalone \
+     --directory /usr/lib/grub/x86_64-efi \
+     --format x86_64-efi \
+     --modules "$MODULES" \
+     --pubkey ./boot.key \
+     --output ./grubx64.efi  \
+     "boot/grub/grub.cfg=./grub.init.cfg" \
+     "boot/grub/grub.cfg.sig=./grub.init.cfg.sig"
+
+    echo "writing signed grub.efi to '$TARGET_EFI'"
+    sudo cp ./grubx64.efi "$TARGET_EFI"
+
+
+.. rst-class:: numbered-step
+
+Prepare grub.cfg
+****************
+
+Define the menu entry for your system in a new GRUB configuration :file: `grub.cfg`.
```

Comment:
Wouldn't that file be overwritten by someone doing a `sudo update-grub`?

Reply:
Will not, path/to/this/grub.cfg will not be touched by 'update-grub', unless user intend to do so, e.g. put it under /boot/grub/

Reply:
OK, thanks!